packages/components/src/slot-fill/index.js

```diff
@@ -1,13 +1,28 @@
 /**
  * Internal dependencies
  */
-import Slot from './slot';
-import Fill from './fill';
-import Provider, { Consumer } from './context';
+import BaseSlot from './slot';
+import BaseFill from './fill';
+import Provider from './context';
+import BubblesVirtuallySlot from './bubbles-virtually/slot';
+import BubblesVirtuallyFill from './bubbles-virtually/fill';
+import useSlot from './bubbles-virtually/use-slot';
 
-export { Slot };
-export { Fill };
-export { Provider, Consumer };
+export function Slot( { bubblesVirtually, ...props } ) {
+	if ( bubblesVirtually ) {
+		return <BubblesVirtuallySlot { ...props } />;
+	}
+	return <BaseSlot { ...props } />;
+}
+
+export function Fill( props ) {
+	return (
+		<>
+			<BaseFill { ...props } />
+			<BubblesVirtuallyFill { ...props } />
+		</>
+	);
+}
```

Comment:
This probably needs some code comments. The reason I'm including both `Fill`s there is because we have to register them even before their respective slot is registered (so `BlockInspector` can check if there are fills before rendering the slot, for example).

They both registers themselves and return `null` if there's no slot. So, one of them will be always `null`.